--- a/src/xtmnanovg.c
+++ b/src/xtmnanovg.c
@@ -22,9 +22,17 @@ int nvgInit(void){
 }

 #elif __linux__
-#include <GL/gl.h>
+#include <stdio.h>
+#include <GL/glew.h>

 int nvgInit(){
+  glewExperimental = GL_TRUE;
+  if(glewInit() != GLEW_OK) {
+    printf("Could not init glew.\n");
+    return GL_FALSE;
+  }
+  // GLEW generates GL error because it calls glGetString(GL_EXTENSIONS), we'll consume it here.
+  glGetError();
   // we don't use GLEW on Linux, so init doesn't do anything
   return GL_TRUE;
 }
