--- a/pm_linux/finddefault.c
+++ b/pm_linux/finddefault.c
@@ -6,6 +6,7 @@

 #include <stdlib.h>
 #include <stdio.h>
+#include <ctype.h>
 #include <string.h>
 #include "portmidi.h"

--- a/pm_linux/pmlinux.c
+++ b/pm_linux/pmlinux.c
@@ -18,6 +18,10 @@ extern void pm_term(void);
 extern PmDeviceID pm_default_input_device_id;
 extern PmDeviceID pm_default_output_device_id;

+/* Forward declaration for find_default_device */
+PmDeviceID pm_find_default_device(char *pattern, int input);
+PmDeviceID find_default_device(char *path, int input);
+
 // (amalg) defined in pmlinux.c:
 void pm_init(void);
 void pm_term(void);